libkbfs/folder_branch_ops.go

```diff
@@ -2248,14 +2246,13 @@ func (fbo *folderBranchOps) CreateFile(
 		entryType = File
 	}
 
-	if excl == WithEXCL {
-		if err = fbo.cr.Wait(ctx); err != nil {
-			return nil, EntryInfo{}, err
-		}
-	}
-
 	err = fbo.doMDWriteWithRetryUnlessCanceled(ctx,
 		func(lState *lockState) error {
+			if excl == WithExcl {
+				if err = fbo.cr.Wait(ctx); err != nil {
```

Comment:
Putting it here is problematic because you will be holding the lock over the Wait call.  If the conflict resolver needs the lock to finalize the resolution, it will deadlock.

It might be better to do it right after this call: https://github.com/keybase/kbfs/blob/master/libkbfs/folder_branch_ops.go#L2225

Obviously would be nice to have a test that covers it as well.  I guess you could use `parallel` and a staller to cook something up, but it might be messy and not worth doing?